pyproject.toml

```diff
@@ -1,30 +1,192 @@
-# It's the equivalent of r-strings in Python.  Multiline strings are treated as
-# verbose regular expressions by Black.  Use [ ] to denote a significant space
-# character.
-
-[tool.black]
-line-length = 100
-target-version = ['py312']
-include = '\.pyi?$'
-exclude = '''
-/(
-    \.git
-  | \.hg
-  | \.mypy_cache
-  | \.tox
-  | \.venv
-  | _build
-  | buck-out
-  | build
-  | dist
-
-  # The following are specific to Black, you probably don't want those.
-  | blib2to3
-  | tests/data
-  | profiling
-)/
-'''
-skip-string-normalization = true
+[build-system]
+requires = ["hatchling", "hatch-vcs"]
+build-backend = "hatchling.build"
+
+
+[project]
+name = "papermill"
+description = "Parameterize and run Jupyter and nteract Notebooks"
+readme = "README.md"
+license = "BSD-3-clause"
+requires-python = ">=3.8"
+authors = [
+    { name = "nteract contributors", email = "nteract@googlegroups.com" },
+]
+keywords = [
+    "jupyter",
+    "mapreduce",
+    "notebook",
+    "nteract",
+    "pipeline",
+]
+classifiers = [
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: System Administrators",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+    "Programming Language :: Python :: 3.12",
+]
+dependencies = [
+    "aiohttp==3.9.0b0;python_version==\"3.12\"",
+    "click",
+    "entrypoints",
+    "nbclient >= 0.2.0",
+    "nbformat >= 5.1.2",
+    "pyyaml",
+    "requests",
+    "tenacity >= 5.0.2",
+    "tqdm >= 4.32.2",
+]
+dynamic = ["version"]
+
+[project.scripts]
+papermill = "papermill.__main__:papermill"
+
+[project.urls]
+Documentation = "https://papermill.readthedocs.io"
+Funding = "https://nteract.io"
+Homepage = "https://github.com/nteract/papermill"
+Source = "https://github.com/nteract/papermill/"
+Tracker = "https://github.com/nteract/papermill/issues"
+
+[tool.hatch.version]
+source = "vcs"
+
+[tool.hatch.build.hooks.vcs]
+version-file = "papermill/version.py"
+
+[tool.hatch.build.targets.sdist]
+include = [
+    "/papermill",
+]
+
+[[tool.hatch.envs.default.matrix]]
+python = ["3.8", "3.9", "3.10", "3.11", "3.12"]
+
+[tool.hatch.envs.default]
+dependencies = [
+    "attrs>=17.4.0",
+    "azure-datalake-store >= 0.0.30",
+    "azure-identity>=1.3.1",
+    "azure-storage-blob >= 12.1.0",
+    "black >= 23.11.0",
+    "boto3",
+    "boto3",
+    "botocore",
+    "bumpversion",
+    "check-manifest",
+    "codecov",
+    "coverage",
+    "flake8",
+    "gcsfs>=0.2.0",
+    "google_compute_engine",
+    "ipython>=5.0",
+    "ipywidgets",
+    "moto",
+    "notebook",
+    "pip>=18.1",
+    "pre-commit",
+    "pytest-cov>=2.6.1",
+    "pytest-env>=0.6.2",
+    "pytest-mock>=1.10",
+    "pytest>=4.1",
```

Comment:
seems like a mix of develop and base requirement, I doubt pre-commit is required for this Project to run :)